uberjar: second build fails

**Describe the bug**
With uberjar enabled, the second time I run `mvn install` it fails with:
```
[ERROR]     [error]: Build step io.quarkus.deployment.pkg.steps.JarResultBuildStep#buildUberJar threw an exception: java.lang.IllegalStateException: java.nio.file.FileAlreadyExistsException: META-INF/services/org.eclipse.microprofile.context.spi.ContextManagerProvider
[ERROR]     at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:949)
[ERROR]     at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
[ERROR]     at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR]     at java.lang.Thread.run(Thread.java:748)
[ERROR]     at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.nio.file.FileAlreadyExistsException: META-INF/services/org.eclipse.microprofile.context.spi.ContextManagerProvider
[ERROR]     at com.sun.nio.zipfs.ZipFileSystem.newOutputStream(ZipFileSystem.java:516)
[ERROR]     at com.sun.nio.zipfs.ZipPath.newOutputStream(ZipPath.java:790)
[ERROR]     at com.sun.nio.zipfs.ZipFileSystemProvider.newOutputStream(ZipFileSystemProvider.java:285)
[ERROR]     at java.nio.file.Files.newOutputStream(Files.java:216)
[ERROR]     at io.quarkus.deployment.pkg.steps.JarResultBuildStep.copyCommonContent(JarResultBuildStep.java:407)
[ERROR]     at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildUberJar(JarResultBuildStep.java:230)
[ERROR]     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]     at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]     at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:940)
[ERROR]     ... 7 more
```

`mvn clean install` works as usual.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 5.2.17-200.fc30.x86_64 #1 SMP Mon Sep 23 13:42:32 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "1.8.0_222"
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 0.26.1

**Additional context**
The build also outputs a lot of warnings like:
```
[WARNING] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Duplicate entry META-INF/quarkus-extension.json entry from io.quarkus:quarkus-resteasy-jackson::jar:0.26.1(compile) will be ignored. Existing file was provided by io.quarkus:quarkus-resteasy-common::jar:0.26.1(compile)
```
This should be avoided.

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
@@ -1,6 +1,9 @@
 package io.quarkus.deployment.pkg.steps;
 
 import static io.quarkus.bootstrap.util.ZipUtils.wrapForJDK8232879;
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.nio.file.StandardOpenOption.WRITE;
 
 import java.io.ByteArrayOutputStream;
 import java.io.FileOutputStream;
@@ -12,6 +15,7 @@ import java.nio.file.FileSystem;
 import java.nio.file.FileVisitOption;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
+import java.nio.file.OpenOption;
 import java.nio.file.Path;
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.StandardCopyOption;
@@ -101,6 +105,9 @@ public class JarResultBuildStep {
             "LICENSE")));
 
     private static final Logger log = Logger.getLogger(JarResultBuildStep.class);
+    // we shouldn't have to specify these flags when opening a ZipFS (since they are the default ones), but failure to do so
+    // makes a subsequent uberJar creation fail in java 8 (but works fine in Java 11)
+    private static final OpenOption[] DEFAULT_OPEN_OPTIONS = { TRUNCATE_EXISTING, WRITE, CREATE };
 
     @BuildStep(onlyIf = ThinJarRequired.class)
     ArtifactResultBuildItem thinJarOutput(ThinJarBuildItem thinJarBuildItem) {
@@ -389,7 +396,7 @@ public class JarResultBuildStep {
             for (TransformedClassesBuildItem.TransformedClass i : transformed) {
                 Path target = runnerZipFs.getPath(i.getFileName());
                 handleParent(runnerZipFs, i.getFileName(), seen);
-                try (final OutputStream out = wrapForJDK8232879(Files.newOutputStream(target))) {
+                try (final OutputStream out = wrapForJDK8232879(Files.newOutputStream(target, DEFAULT_OPEN_OPTIONS))) {
                     out.write(i.getData());
                 }
                 seen.put(i.getFileName(), "Current Application");
@@ -403,7 +410,7 @@ public class JarResultBuildStep {
             if (Files.exists(target)) {
                 continue;
             }
-            try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(target))) {
+            try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(target, DEFAULT_OPEN_OPTIONS))) {
                 os.write(i.getClassData());
             }
         }
@@ -417,7 +424,7 @@ public class JarResultBuildStep {
             if (i.getName().startsWith("META-INF/services")) {
                 services.computeIfAbsent(i.getName(), (u) -> new ArrayList<>()).add(i.getClassData());
             } else {
-                try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(target))) {
+                try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(target, DEFAULT_OPEN_OPTIONS))) {
                     os.write(i.getClassData());
                 }
             }
@@ -426,7 +433,8 @@ public class JarResultBuildStep {
         copyFiles(appArchives.getRootArchive().getArchiveRoot(), runnerZipFs, services);
 
         for (Map.Entry<String, List<byte[]>> entry : services.entrySet()) {
-            try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(runnerZipFs.getPath(entry.getKey())))) {
+            try (final OutputStream os = wrapForJDK8232879(
+                    Files.newOutputStream(runnerZipFs.getPath(entry.getKey()), DEFAULT_OPEN_OPTIONS))) {
                 for (byte[] i : entry.getValue()) {
                     os.write(i);
                     os.write('\n');
@@ -504,7 +512,7 @@ public class JarResultBuildStep {
             }
         }
         attributes.put(Attributes.Name.MAIN_CLASS, config.mainClass);
-        try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(manifestPath))) {
+        try (final OutputStream os = wrapForJDK8232879(Files.newOutputStream(manifestPath, DEFAULT_OPEN_OPTIONS))) {
             manifest.write(os);
         }
     }

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/PackageIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/PackageIT.java
@@ -47,6 +47,12 @@ public class PackageIT extends MojoTestBase {
             throws MavenInvocationException, FileNotFoundException, InterruptedException {
         testDir = initProject("projects/uberjar-check", "projects/project-uberjar-true");
 
+        createAndVerifyUberJar();
+        // ensure that subsequent package without clean also works
+        createAndVerifyUberJar();
+    }
+
+    private void createAndVerifyUberJar() throws FileNotFoundException, MavenInvocationException, InterruptedException {
         running = new RunningInvoker(testDir, false);
         final MavenProcessInvocationResult result = running.execute(Collections.singletonList("package"),
                 Collections.singletonMap("QUARKUS_PACKAGE_TYPES", "uber-jar"));

